### PR TITLE
CI workflow

### PR DESCRIPTION
**Pull request by jf514**
Implemented CI via GitHub Actions:
- Added `ci.yml` + other package related files (`pyproject.toml`, `__init__.py`, `version.py`, `codecov.yml`)
- Fixed 60+ pydocs violations
- Fixed black violations

NOTE: Many docstrings added were minimally descriptive. Issue [24](https://github.com/talmolab/stac-mjx/issues/24) was created to address this. 

## Summary by CodeRabbit

- **New Features**
  - Introduced a Continuous Integration (CI) workflow for automated code quality checks and testing.
  - Added a new configuration file for coverage reporting, ensuring code quality metrics are maintained.

- **Documentation**
  - Updated README with revised command for running scripts.
  - Enhanced docstrings across various modules, improving clarity and context for function purposes.

- **Chores**
  - Added dependencies for code formatting and testing tools to improve development environment.
  - Created a dummy test to verify the CI pipeline functionality.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# Continuous integration
+name: CI
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize]
+    paths:
+      - "core/**"
```
Comment (talmo) [inline .github/workflows/ci.yml:8]:
```suggestion
      - "stac_mjx/**"
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      - "tests/**"
+      - ".github/workflows/ci.yml"
+      - "environment.yml"
+  push:
+    branches:
+      - main
+    paths:
+      - "core/**"
```
Comment (talmo) [inline .github/workflows/ci.yml:16]:
```suggestion
      - "stac_mjx/**"
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      with:
+        python-version: "3.12"
+
+    - name: Install dependencies
+      run: |
+        pip install --editable .[dev]
+
+    - name: Run Black
+      run: |
+        black --check core tests
```
Comment (talmo) [inline .github/workflows/ci.yml:49]:
```suggestion
        black --check stac_mjx tests
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    runs-on: "ubuntu-latest"
+    steps:
+
+    - name: Checkout repo
+      uses: actions/checkout@v4
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.12"
```
Comment (talmo) [inline .github/workflows/ci.yml:41]:
```suggestion
        python-version: "3.11"
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        pydocstyle --convention=google core/
+
+  # Tests with pytest
+  tests:
+    timeout-minutes: 15
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest", "windows-latest", "macos-14"]
+        python: ["3.8", "3.12"]
```
Comment (talmo) [inline .github/workflows/ci.yml:62]:
```suggestion
        python: ["3.11"]
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+      run: |
+        pydocstyle --convention=google core/
+
+  # Tests with pytest
+  tests:
+    timeout-minutes: 15
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest", "windows-latest", "macos-14"]
```
Comment (talmo) [inline .github/workflows/ci.yml:61]:
```suggestion
        os: ["ubuntu-latest"]
```

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+      run: |
+        pip install --editable .[dev]
+
+    - name: Run Black
+      run: |
+        black --check core tests
+    
+    - name: Run pydocstyle
+      run: |
+        pydocstyle --convention=google core/
```
Comment (talmo) [inline .github/workflows/ci.yml:53]:
```suggestion
        pydocstyle --convention=google stac_mjx/
```

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+          conda info
+          conda list
+          pip freeze
+
+      - name: Install graphics dependencies on Ubuntu
+        # https://github.com/conda-forge/opencv-feedstock/issues/401
+        if: ${{ startsWith(matrix.os, 'ubuntu') }}
+        shell: bash -l {0}
+        run: |
+          sudo apt-get update && sudo apt-get install libglapi-mesa libegl-mesa0 libegl1 libopengl0 libgl1-mesa-glx
```
Comment (talmo) [inline .github/workflows/ci.yml:95]: Not needed here

File: .github/workflows/ci.yml [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+      - name: Test with pytest (with coverage)
+        shell: bash -l {0}
+        run: |
+          pytest --cov=sleap_io --cov-report=xml --durations=-1 tests/
```
Comment (talmo) [inline .github/workflows/ci.yml:100]:
```suggestion
          pytest --cov=stac_mjx --cov-report=xml --durations=-1 tests/
```

File: pyproject.toml [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+authors = [
+    {name = "Talmo Pereira", email = "talmo@salk.edu"}]
+description="TBD"
+requires-python = ">=3.11"
+keywords = ["TBD"]
+license = {text = "BSD-3-Clause"}
+classifiers = [
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
```
Comment (talmo) [inline pyproject.toml:16]: Probably take out <3.10

File: pyproject.toml [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+    "numpy",
+    "attrs",
+    "h5py>=3.8.0",
+    "pynwb",
+    "ndx-pose",
+    "pandas",
+    "simplejson",
+    "imageio",
+    "imageio-ffmpeg",
+    "av"]
```
Comment (talmo) [inline pyproject.toml:30]: Replace this with the actual dependencies

File: pyproject.toml [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+    "black",
+    "pydocstyle",
+    "toml",
+    "twine",
+    "build",
+    "mkdocs-material[imaging]",
+    "mkdocs-jupyter",
+    "mkdocstrings[python]>=0.18",
+    "mkdocs-gen-files",
+    "mkdocs-literate-nav"]
```
Comment (talmo) [inline pyproject.toml:51]: These are dev-only dependencies, don't need the mkdocs ones

File: pyproject.toml [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+
+#[project.urls]
+#Homepage = ""
+#Repository = "https://github.com/talmolab/sleap-io"
```
Comment (talmo) [inline pyproject.toml:55]:
```suggestion
[project.urls]
Homepage = "https://github.com/talmolab/stac-mjx"
Repository = "https://github.com/talmolab/stac-mjx"
```

File: pyproject.toml [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+
+[pydocstyle]
+convention = "google"
+match-dir = "stac-mjx"
+
+[tool.coverage.run]
+source = ["livecov"]
+
+[tool.pytest.ini_options]
+addopts = "--cov sleap_io --cov-report=lcov:lcov.info --cov-report=term"
```
Comment (talmo) [inline pyproject.toml:71]:
```suggestion
addopts = "--cov stac_mjx --cov-report=lcov:lcov.info --cov-report=term"
```

File: pyproject.toml [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+
+[tool.setuptools.packages.find]
+exclude = ["site"]
+
+[tool.black]
+line-length = 88
+
+[pydocstyle]
+convention = "google"
+match-dir = "stac-mjx"
```
Comment (talmo) [inline pyproject.toml:65]:
```suggestion
match-dir = "stac_mjx"
```

File: pyproject.toml [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+    "ndx-pose",
+    "pandas",
+    "simplejson",
+    "imageio",
+    "imageio-ffmpeg",
+    "av"]
+dynamic = ["version", "readme"]
+
+[tool.setuptools.dynamic]
+version = {attr = "stac-mjx.version.__version__"}
```
Comment (talmo) [inline pyproject.toml:34]:
```suggestion
version = {attr = "stac_mjx.version.__version__"}
```

**Review by talmo — CHANGES_REQUESTED** [review]
Rename `core/` -> `stac_mjx/`. This is needed since python module names can't have hyphen (`-`) so we replace that with `_`. The **package** name is still `stack-mjx`.

File: README.md [hunk 17]
```diff
@@ -16,7 +16,7 @@ conda activate stac-mjx-env
 
 2. For new data, first run stac on just a small subset of the data with
 
-    `python core/main.py test.skip_transform=True`
+    `python stac-mjx/main.py test.skip_transform=True`
```
Comment (talmo) [inline README.md:19]:
```suggestion
    `python stac_mjx/main.py test.skip_transform=True`
```

File: videos/direct_render.mp4
Comment (talmo) [inline videos/direct_render.mp4:1]: Is this supposed to be in the PR?

File: viz_usage.ipynb
Comment (talmo) [inline viz_usage.ipynb:1]: Is this supposed to be in the PR?

File: version.py
Comment (talmo) [inline version.py:1]: Delete this file (we only use the one in the package, i.e., `stac_mjx/version.py`)

**Review by talmo — CHANGES_REQUESTED** [review]

File: videos/direct_render.mp4
Comment (jf514) [reply videos/direct_render.mp4:1]: No.

File: viz_usage.ipynb
Comment (jf514) [reply viz_usage.ipynb:1]: Yes. References `stac_mjx` directory.

**Review by talmo — APPROVED** [review]
Ok, this LGTM so I'll approve but I think we might still have a couple of dangling files that should be removed if they're not used:

- `setup_new.py`
- `viz_usage.ipynb`

**Pull request merged**